DictionaryConsoleApp/Program.cs

```diff
@@ -0,0 +1,153 @@
+            Exercise1();
+            Exercise2();
+            Exercise3();
+            Exercise4();
+        }
+
+        public static void Exercise1()
+        {
+            Console.WriteLine("Exercise 1.");
+
+            var employees = new Dictionary<Student, int>
+            {
+                { new Student(4, "John"), 1},
+                { new Student(5, "Henry"), 2 }
+            };
+
+            foreach (KeyValuePair<Student, int> pair in employees)
+            {
+                Console.WriteLine($"Student {pair.Key.Name}, estimation: {pair.Key.AverageRating}.");
+            }
+        }
+
+        public static void Exercise2()
+        {
+            var dictionaryEnglishFranch = new Dictionary<string, string>(10)
+            {
+                    { "book", "livre" },
+                    { "mouse", "souris" },
+                    { "dictionary", "dictionnaire" },
+                    { "notepad", "bloc-notes" },
+                    { "word", "mot" },
+                    { "time", "temps" },
+                    { "music", "musique" },
+                    { "game", "jeu" },
+                    { "flower", "fleur" },
+                    { "candles", "bougies" }
+            };
+
+            var input = Console.ReadLine();
+
+            if (input == null)
+            {
+                Console.WriteLine("Enter the word again");
+                Exercise2();
+            }
+
+            Console.WriteLine(GetTranslation(dictionaryEnglishFranch, input));
+
+            string GetTranslation(Dictionary<string, string> dictionaryEnglishFranch, string input)
+            {
+                string targetValue = dictionaryEnglishFranch.FirstOrDefault(pair => pair.Key == input).Value;
+                var output = "";
+
+                if (targetValue != null)
+                {
+                    output = targetValue;
+                    return output;
+                }
+                else
+                {
+                    Console.WriteLine("Key not found. Enter the word again");
+                    Exercise2();
+                    return output;
+                }
+            }
+        }
+
+        public static void Exercise3()
+        {
+            var listOfProducts = new Dictionary<string, int>(10)
+            {
+                    { "book", 500 },
+                    { "mouse", 4300 },
+                    { "dictionary", 3500 },
+                    { "notepad", 990 },
+                    { "keys", 600},
+                    { "alarm сlock", 3400},
+                    { "music record", 1500 },
+                    { "game", 400},
+                    { "flower", 2000 },
+                    { "candles", 400}
+            };
+
+            var keysToRemove = new HashSet<string>() { "book", "keys" };
+
+            var filteredDict = listOfProducts
+                .Where(kvp => !keysToRemove
+                .Contains(kvp.Key))
+                .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
```

Comment:
This is a good filtering method.
But the specification requires a method with which you can delete keys